src/pages/MarketPage/components/AllItemsSection.jsx

```diff
@@ -0,0 +1,113 @@
+import React, { useEffect, useState } from 'react'
+import { getProducts } from '../../../api/itemApi'
+import ItemCard from './ItemCard'
+import { ReactComponent as SortIcon } from '../../../assets/images/icons/ic_sort.svg'
+import { ReactComponent as SearchIcon } from '../../../assets/images/icons/ic_search.svg'
+import { Link } from 'react-router-dom'
+import DropdownList from '../../../components/UI/DropdownList'
+import PaginationBar from '../../../components/UI/PaginationBar'
+
+const getPageSize = () => {
+  const width = window.innerWidth
+  if (width < 768) {
+    // Mobile viewport
+    return 4
+  } else if (width < 1280) {
+    // Tablet viewport
+    return 6
+  } else {
+    // Desktop viewport
+    return 10
+  }
```

Comment:
P3:
early return 문 방식으로 작성되어 있으므로 아래처럼 해도 동일 동작합니다.

```suggestion
  } 
  // Tablet viewport
  if (width < 1280) {
    return 6
  }
  
  // Desktop viewport
  return 10
```